pkgs/applications/misc/cool-retro-term/default.nix

```diff
@@ -16,15 +16,23 @@ stdenv.mkDerivation rec {
     sed -i -e '/qmltermwidget/d' cool-retro-term.pro
   '';
 
-  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols qtgraphicaleffects ];
-  nativeBuildInputs = [ qmake ];
+  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols qtgraphicaleffects qtdeclarative ];
+  nativeBuildInputs = [ qmake makeWrapper ];
 
   installFlags = [ "INSTALL_ROOT=$(out)" ];
 
   preFixup = ''
     mv $out/usr/share $out/share
     mv $out/usr/bin $out/bin
     rmdir $out/usr
+  '' + stdenv.lib.optionalString stdenv.isDarwin ''
+    # Fixes "This application failed to start because it could not find or load the Qt
+    # platform plugin "cocoa"."
+    wrapProgram $out/bin/cool-retro-term.app/Contents/MacOS/cool-retro-term \
+     --set QT_QPA_PLATFORM_PLUGIN_PATH ${qtbase.bin}/lib/qt-*/plugins/platforms \
+     --set QT_PLUGIN_PATH ${qtbase.bin}/${qtbase.qtPluginPrefix} \
+     --set QML2_IMPORT_PATH ${qmltermwidget}/${qtbase.qtQmlPrefix}:${qtdeclarative.bin}/${qtbase.qtQmlPrefix}:${qtgraphicaleffects}/${qtbase.qtQmlPrefix}:${qtquickcontrols}/${qtbase.qtQmlPrefix}
```

Comment:
Is this also the case if you install retro-term in your nix profile?

Reply:
Oh, that works! Also works without wrapper using `nix run`. I have removed the wrapper, it seems like all that is needed is symlink to binary.